tf_encrypted/protocol/pond/pond.py

```diff
@@ -221,7 +221,7 @@ def define_private_placeholder(
     factory = factory or self.tensor_factory
 
     suffix = "-" + name if name else ""
-    with tf.name_scope("private-placeholder{}".format(suffix)):
+    with tf.compat.v1.name_scope("private-placeholder{}".format(suffix)):
```

Comment:
there seems to still be a [`tf.name_scope`](https://www.tensorflow.org/api_docs/python/tf/name_scope) in TF2 although it looks a bit different. did you play around with it to see what it would take to use it instead?

Reply:
It works almost the same way but I think they made some minor changes so it wasn't safe to auto-convert it. Will look into it more to find out what these changes are.

Reply:
any updates here @justin1121?

Reply:
Haven't really found good documentation explaining the differences. The auto-convert script says something about re-entering a scope via the name (v1 name scope implementation) but I don't think we're doing that anywhere anyway.

Reply:
let's keep it for now and revisit later when we remove more of the compatibility functionality